backend/src/main/java/ai/verta/modeldb/versioning/blob/diff_factory/BlobDiffFactory.java

```diff
@@ -0,0 +1,104 @@
+package ai.verta.modeldb.versioning.blob.diff_factory;
+
+import ai.verta.modeldb.versioning.Blob.ContentCase;
+import ai.verta.modeldb.versioning.BlobDiff;
+import ai.verta.modeldb.versioning.BlobDiff.Builder;
+import ai.verta.modeldb.versioning.BlobExpanded;
+import ai.verta.modeldb.versioning.DiffStatusEnum.DiffStatus;
+import com.google.rpc.Code;
+import com.google.rpc.Status;
+import io.grpc.protobuf.StatusProto;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/** constructs proto blob diff object from proto blobs*/
+public abstract class BlobDiffFactory {
+
+  private final BlobExpanded blobExpanded;
+
+  BlobDiffFactory(BlobExpanded blobExpanded) {
+    this.blobExpanded = blobExpanded;
+  }
+
+  private ContentCase getBlobType() {
+    return blobExpanded.getBlob().getContentCase();
+  }
+
+  public BlobExpanded getBlobExpanded() {
+    return blobExpanded;
+  }
+
+  public List<BlobDiff> compare(BlobDiffFactory blobDiffFactory, String location) {
```

Comment:
```suggestion
  public List<BlobDiff> compare(BlobDiffFactory blobDiffFactoryB, String location) {
```
If you are using a name with given suggestion then it will help to understand code better way.